### PR TITLE
[MERGE] rating,project,livechat: refactor API for performance issue

### PR DESCRIPTION
**Pull request by jem-odoo**
This commit makes several improvement in order to speed up and clean
the rating API. No functionnal changes should be obsered.
Multiple point are covered here:

1/ introduces a parent.mixin: since we want to extract the satisfaction
percentage on parent rating model (project, team, livechat channel), 
providing a mixin to standardize this computation in a performant and
lazy way seems a good optimization (more details in the concerned
commit).

2/ cleaning API: some methods needed to be private and/or renamed.

3/ Better integration in the business application: use the new parent
mixin, remove legacy code and fields, ...

Task-1903565

### CODE REVIEW

File: addons/rating/models/rating.py [hunk 1]
```diff
@@ -133,6 +134,44 @@ def action_open_rated_object(self):
...
+            elif rating > RATING_LIMIT_OK:
+                grades_per_parent[parent_id]['okay'] += item['__count']
+            else:
+                grades_per_parent[parent_id]['bad'] += item['__count']
+
+        # compute percentage per parent
+        for record in self:
+            repartition = grades_per_parent.get(record.id)
+            if repartition:
+                record.rating_percentage_satisfaction = repartition['great'] * 100 / sum(repartition.values()) if sum(repartition.values()) else -1
```
Comment (tde-banana-odoo) [inline addons/rating/models/rating.py]: Linting: if sum(repartition.values() could be moved into the "if repartition" condition.

File: addons/im_livechat/models/im_livechat_channel.py [hunk 2]
```diff
@@ -15,7 +15,9 @@ class ImLivechatChannel(models.Model):
     """
 
     _name = 'im_livechat.channel'
+    _inherit = ['rating.parent.mixin']
     _description = 'Livechat Channel'
+    _rating_satisfaction_days = 7  # include only last 7 days to compute satisfaction
```
Comment (tde-banana-odoo) [inline addons/im_livechat/models/im_livechat_channel.py]: Why so 7 ?

File: addons/rating/models/rating.py [hunk 3]
```diff
@@ -133,6 +134,44 @@ def action_open_rated_object(self):
         }
 
 
+class RatingParentMixin(models.AbstractModel):
```
Comment (tde-banana-odoo) [inline addons/rating/models/rating.py]: Linting: we discussed to have another file with this new mixin + the other existing one.

**Comment by jem-odoo** [comment]
@robodoo rebase-merge please !

**Comment by d-fence** [comment]
@robodoo r+

**Pull request closed without merging**